src/app/core/session/currently-logged-in.ts

```diff
@@ -0,0 +1,10 @@
+import { BehaviorSubject } from "rxjs";
+import { Entity } from "../entity/model/entity";
+import { Injectable } from "@angular/core";
+
+@Injectable()
+export class CurrentlyLoggedInSubject extends BehaviorSubject<Entity> {
```

Comment:
this name is missing a bit that you are getting an Entity (not just a boolean state, for example). Why not keep the name "CurrentUserSubject"?